lib/pushmi_pullyu/aip_downloader_files_directories.rb

```diff
@@ -0,0 +1,91 @@
+require 'rdf'
+require 'rdf/n3'
+require 'fileutils'
+
+# Download all of the metadata/datastreams and associated data
+# related to an object
+
+module PushmiPullyu::AipDownloaderFilesDirectories
+  FIXITY_PATH = '/content/fcr:fixity'.freeze
+  CONTENT_DATASTREAM_METADATA_PATH = '/content/fcr:metadata'.freeze
+  CHARACTERIZATION_PATH = '/characterization'.freeze
+  VERSIONS_PATH = '/content/fcr:versions'.freeze
+  CONTENT_PATH = '/content'.freeze
+  THUMBNAIL_PATH = '/thumbnail'.freeze
+  FEDORA3FOXML_PATH = '/fedora3foxml'.freeze
+  FEDORA3FOXML_METADATA_PATH = '/fedora3foxml/fcr:metadata'.freeze
+
+  FILENAME_PREDICATE = 'info:fedora/fedora-system:def/model#downloadFilename'.freeze
+
+  def self.included(_klass)
+    attr_reader :workdir, :basedir, :datadir, :objectsdir, :metadatadir,
+                :logsdir, :thumbnailsdir,
+                :tar_filename, :main_object_filename, :fixity_report_filename,
+                :content_datastream_metadata_filename,
+                :characterization_filename, :versions_filename,
+                :thumbnail_filename, :fedora3foxml_filename,
+                :fedora3foxml_metadata_filename
+  end
+
+  def make_object_directories
+    clean_directories
+    logger.debug("#{noid}: Creating directories ...")
```

Comment:
Yeah, this is the application logger (not the logger that will produce a creation log for the current AIP). The AIP creation log can't be used at this point, since the directory it will live in doesn't exist yet. (Maybe these logging statements aren't needed, they are set to 'debug')